source/blog/2018-07-19-ovn-dpdk.html.md

```diff
@@ -0,0 +1,78 @@
+# ansible-galaxy install oVirt.dpdk-setup
+```
+An example playbook:
+```
+- hosts: dpdk_host_0
+  vars:
+    pci_drivers:
+      "0000:02:00.1": "vfio-pci"
+      "0000:02:00.2": "igb"
+      "0000:02:00.3": ""
+    configure_kernel: true
+    bind_drivers: true
+    set_ovs_dpdk: false
+  roles:
+    - ovirt-ansible-dpdk-setup
+```
+The role is controlled by 3 boolean variables (all set to `true` by default) and a dictionary of devices and their drivers:
+* `configure_kernel` -- determines whether the kernel should be configured for DPDK usage (hugepages, CPU partitioning)
+* `bind_drivers` -- determines whether the role should bind devices to their specified drivers.
+* `set_ovs_dpdk` -- determines whether the role should initialize and configure OVS for DPDK usage.
+* `pci_drivers` -- a dictionary of the PCI addresses of network interface cards as keys and drivers as values. Empty strings represent the kernel default. non-DPDK compatible drivers won't be configured for OVS-DPDK usage.
+
+Additionally, there are several optional performance related variables:
+* `pmd_threads_count` -- determines the amount of PMD threads per NIC (default: `1`).
+* `nr_2mb_hugepages` -- determines the amount of 2MB hugepages, if 2MB hugepages are to be used (default: `1024`).
+* `nr_1gb_hugepages` -- determines the amount of 1GB hugepages, if 1GB hugepages are to be used (default: `4`).
+* `use_1gb_hugepages` -- determines whether 1GB hugepages should be used, where 1GB hugepages are supported (default: `true`).
+
+For additional information, refer to the role's [repository](https://github.com/ovirt/ovirt-ansible-dpdk-setup/).
+
+### OVN Integration
+DPDK in oVirt now leverages the capabilities of OVN in the form of an OVN localnet. This allows for seamless connectivity across OVN networks, benefiting from OVN's software defined routers, switches, security groups, and ACL's.
+
+For more information about OVN localnet integration in oVirt, refer to oVirt's [Provider Physical Network RFE](https://ovirt.org/develop/release-management/features/network/provider-physical-network/).
+
+# Usage in oVirt
+1) Install `oVirt.dpdk-setup` Ansible role via ansible-galaxy:
+    ```
+    # ansible-galaxy install oVirt.dpdk-setup
+    ```
+2) Execute the role as described above.
+3) Unless the host was already configured, the host will restart for kernel changes to be applied.
+4) Make sure the following:
+    * IOMMU: ```/sys/devices/virtual/iommu/<DMAR device>/devices/<PCI address>``` should exist. DMAR devices are typically annotated by dmar0, dmar1, and so forth.
+    * hugepages: ```grep Huge /proc/meminfo``` should reflect the desired state as defined in the Ansible playbook (i.e. `Hugepagesize` and `HugePages_Total`)
+    * CPU partitioning: based on the devices that are to be used with a DPDK compatible driver, CPU cores should be partitioned separately on each NUMA node. Refer to `lscpu` to see live CPU NUMA separation information, e.g.: `NUMA node1 CPU(s):     8-15`
```

Comment:
you ask to "make sure", but what should I do if it does not? is it likely to fail?

maybe we'd like to use a different language, such "note that after reboot the following values are changed"
If it does not, you probably have an incompatible hardware or should report a bug on user@ovirt.org.

Reply:
Done